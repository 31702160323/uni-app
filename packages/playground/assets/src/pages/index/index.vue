<template>
    <view>
        <test-com></test-com>
        <image src="./static/invalid-static.jpg" />
        <image src="../../static/logo.png" />
    </view>
</template>
<script>
    export default {
        onLoad() {
        }
    }
</script>