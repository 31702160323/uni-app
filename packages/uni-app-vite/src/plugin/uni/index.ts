import { isNativeTag, isCustomElement } from '@dcloudio/uni-shared'
import { compileI18nJsonStr } from '@dcloudio/uni-i18n'
import { UniVitePlugin, initI18nOptions } from '@dcloudio/uni-cli-shared'

export function uniOptions(): UniVitePlugin['uni'] {
  return {
    copyOptions() {
      const platfrom = process.env.UNI_PLATFORM
      const inputDir = process.env.UNI_INPUT_DIR
      const outputDir = process.env.UNI_OUTPUT_DIR
      return {
        assets: ['hybrid/html/**/*', 'uni_modules/*/hybrid/html/**/*'],
        targets: [
          {
            src: 'androidPrivacy.json',
            dest: outputDir,
            transform(source) {
              const options = initI18nOptions(platfrom, inputDir, false, true)
              if (!options) {
                return
              }
              return compileI18nJsonStr(source.toString(), options)
            },
          },
        ],
      }
    },
    compilerOptions: {
      isNativeTag,
      isCustomElement,
    },
    transformEvent: {
      tap: 'click',
    },
  }
}
