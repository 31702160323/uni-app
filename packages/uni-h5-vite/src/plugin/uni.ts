import {
  transformH5BuiltInComponents,
  transformMatchMedia,
  transformPageHead,
  transformTapToClick,
  UniVitePlugin,
} from '@dcloudio/uni-cli-shared'
import { isH5NativeTag, isH5CustomElement } from '@dcloudio/uni-shared'
import { CompilerOptions } from '@vue/compiler-core'

export const compilerOptions: CompilerOptions = {
  isNativeTag: isH5NativeTag,
  isCustomElement: isH5CustomElement,
  nodeTransforms: [
    transformH5BuiltInComponents,
    transformTapToClick,
    transformMatchMedia,
    transformPageHead,
  ],
}

export function createUni(): UniVitePlugin['uni'] {
  return {
    copyOptions: {
      assets: ['hybrid/html'],
    },
    compilerOptions,
  }
}
