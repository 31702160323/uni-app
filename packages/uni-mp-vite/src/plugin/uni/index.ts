import { isNativeTag, isCustomElement } from '@dcloudio/uni-shared'
import { EmittedFile } from 'rollup'
import { CopyOptions, UniVitePlugin } from '@dcloudio/uni-cli-shared'
import { TemplateCompiler } from '@vue/compiler-sfc'

import * as compiler from '@dcloudio/uni-mp-compiler'

export function uniOptions({
  copyOptions,
  miniProgram,
}: {
  copyOptions: CopyOptions
  miniProgram: {
    slot: {
      fallback: boolean
    }
    filter?: {
      lang: string
    }
    directive: string
    emitFile?: (emittedFile: EmittedFile) => string
  }
}): UniVitePlugin['uni'] {
  return {
    copyOptions,
    compiler: compiler as TemplateCompiler,
    compilerOptions: {
      miniProgram,
      isNativeTag,
      isCustomElement,
    },
  }
}
