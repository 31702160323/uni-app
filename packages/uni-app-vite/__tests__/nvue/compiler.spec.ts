import {
  DirectiveNode,
  ElementNode,
  findDir,
  findProp,
  SimpleExpressionNode,
} from '@vue/compiler-core'
import { compileTemplate } from '@vue/compiler-sfc'
import { uniOptions } from '../../src/plugin/uni/index'
process.env.UNI_COMPILER = 'nvue'
const { compilerOptions } = uniOptions()!
const filename = 'foo.vue'

function compile(source: string) {
  return compileTemplate({
    source,
    filename,
    id: filename,
    compilerOptions: {
      ...compilerOptions,
    },
  }).ast!.children[0] as ElementNode
}
describe('app-nvue: compiler', () => {
  test('u-tags', () => {
    ;[
      'text',
      'image',
      'input',
      'textarea',
      'video',
      'web-view',
      'slider',
    ].forEach((tag) => {
      expect(compile(`<${tag}></${tag}>`).tag).toBe(`u-${tag}`)
    })
  })
  test('video', () => {
    expect(compile(`<video></video>`).children.length).toBe(0)
    expect(
      (compile(`<video><view></view></video>`).children[0] as ElementNode).tag
    ).toBe('u-scalable')
  })
  test('scroll-view', () => {
    compile(`<view></view>`)
  })
  test('render-whole', () => {
    expect(
      (
        (
          findProp(
            compile(`<view :render-whole="true">hello</view>`),
            'appendAsTree',
            true,
            false
          ) as DirectiveNode
        ).arg as SimpleExpressionNode
      ).content
    ).toBe('appendAsTree')
  })
  test('unitary tag', () => {
    expect(
      findProp(compile(`<text>hello</text>`), 'appendAsTree', true, false)
    ).toBeTruthy()
  })
  test('tap=>click', () => {
    expect(
      (
        findDir(compile(`<view @tap="click"></view>`), 'on')!
          .arg as SimpleExpressionNode
      ).content
    ).toBe('click')
  })
})
