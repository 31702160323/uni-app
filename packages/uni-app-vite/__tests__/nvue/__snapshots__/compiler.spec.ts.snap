// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app-nvue: compiler <video></video> 1`] = `
"import { openBlock as _openBlock, createElementBlock as _createElementBlock } from \\"vue\\"

export function render(_ctx, _cache) {
  return (_openBlock(), _createElementBlock(\\"u-video\\"))
}"
`;

exports[`app-nvue: compiler <video><view></view></video> 1`] = `
"import { createElementVNode as _createElementVNode, openBlock as _openBlock, createElementBlock as _createElementBlock } from \\"vue\\"

const _hoisted_1 = /*#__PURE__*/_createElementVNode(\\"u-scalable\\", { style: {position:\\"absolute\\",left:\\"0\\",right:\\"0\\",top:\\"0\\",bottom:\\"0\\"} }, [
  /*#__PURE__*/_createElementVNode(\\"view\\")
], -1 /* HOISTED */)
const _hoisted_2 = [
  _hoisted_1
]

export function render(_ctx, _cache) {
  return (_openBlock(), _createElementBlock(\\"u-video\\", null, _hoisted_2))
}"
`;

exports[`app-nvue: compiler <view><text>hello</text></view> 1`] = `
"import { createElementVNode as _createElementVNode, openBlock as _openBlock, createElementBlock as _createElementBlock } from \\"vue\\"

const _hoisted_1 = [\\"appendAsTree\\"]

export function render(_ctx, _cache) {
  return (_openBlock(), _createElementBlock(\\"view\\", null, [
    _createElementVNode(\\"u-text\\", { appendAsTree: true }, \\"hello\\", 8 /* PROPS */, _hoisted_1)
  ]))
}"
`;

exports[`app-nvue: compiler <view>hello</view> 1`] = `
"import { createElementVNode as _createElementVNode, openBlock as _openBlock, createElementBlock as _createElementBlock } from \\"vue\\"

const _hoisted_1 = [\\"appendAsTree\\"]

export function render(_ctx, _cache) {
  return (_openBlock(), _createElementBlock(\\"view\\", null, [
    _createElementVNode(\\"u-text\\", { appendAsTree: true }, \\"hello\\", 8 /* PROPS */, _hoisted_1)
  ]))
}"
`;

exports[`app-nvue: compiler <view>hello{{a}}<view>aaa{{a}}</view>{{b}}</view> 1`] = `
"import { createElementVNode as _createElementVNode, toDisplayString as _toDisplayString, openBlock as _openBlock, createElementBlock as _createElementBlock } from \\"vue\\"

const _hoisted_1 = [\\"appendAsTree\\"]
const _hoisted_2 = [\\"appendAsTree\\"]
const _hoisted_3 = [\\"appendAsTree\\"]
const _hoisted_4 = [\\"appendAsTree\\"]
const _hoisted_5 = [\\"appendAsTree\\"]

export function render(_ctx, _cache) {
  return (_openBlock(), _createElementBlock(\\"view\\", null, [
    _createElementVNode(\\"u-text\\", { appendAsTree: true }, \\"hello\\", 8 /* PROPS */, _hoisted_1),
    _createElementVNode(\\"u-text\\", { appendAsTree: true }, _toDisplayString(_ctx.a), 9 /* TEXT, PROPS */, _hoisted_2),
    _createElementVNode(\\"view\\", null, [
      _createElementVNode(\\"u-text\\", { appendAsTree: true }, \\"aaa\\", 8 /* PROPS */, _hoisted_3),
      _createElementVNode(\\"u-text\\", { appendAsTree: true }, _toDisplayString(_ctx.a), 9 /* TEXT, PROPS */, _hoisted_4)
    ]),
    _createElementVNode(\\"u-text\\", { appendAsTree: true }, _toDisplayString(_ctx.b), 9 /* TEXT, PROPS */, _hoisted_5)
  ]))
}"
`;
